composer/callbacks/speed_monitor.py

```diff
@@ -284,22 +285,37 @@ def batch_end(self, state: State, logger: Logger):
             except AttributeError:
                 pass
 
-            composer_model = state.model
-            if not isinstance(composer_model, ComposerModel):
-                composer_model = composer_model.module  # Pass through DDP wrapping
-            if hasattr(composer_model, 'flops_per_batch'):
-                model_flops_per_batch = composer_model.flops_per_batch  # type: ignore
-                if not isinstance(model_flops_per_batch, Callable):
-                    raise TypeError('flops_per_batch must a callable accepting a batch and '
-                                    f'returning an int or float. Instead, got {type(model_flops_per_batch)}.')
-                flops_per_batch = model_flops_per_batch(state.batch)
-                flops_per_sec = flops_per_batch * batches_per_sec
-                logger.log_metrics({'throughput/flops_per_sec': flops_per_sec})
-                dev_flops_per_sec = flops_per_sec / world_size
-                logger.log_metrics({'throughput/device/flops_per_sec': dev_flops_per_sec})
-                if self.gpu_flops_available:
-                    mfu = dev_flops_per_sec / self.gpu_flops_available
-                    logger.log_metrics({'throughput/device/mfu': mfu})
+        # Compute flops stats if model has flops_per_batch
+        composer_model = state.model
+        if not isinstance(composer_model, ComposerModel):
+            composer_model = composer_model.module  # Pass through DDP wrapping
+        if hasattr(composer_model, 'flops_per_batch'):
+            model_flops_per_batch = composer_model.flops_per_batch  # type: ignore
+            if not isinstance(model_flops_per_batch, Callable):
+                raise TypeError('flops_per_batch must a callable accepting a batch and '
+                                f'returning an int or float. Instead, got {type(model_flops_per_batch)}.')
+            device_flops_per_batch = model_flops_per_batch(state.batch)
+
+            # Sum flops across all ranks since each rank computes the flops for its own batch
+            flops_per_batch_tensor = state.device.tensor_to_device(
+                torch.tensor(device_flops_per_batch, dtype=torch.float))
+            dist.all_reduce(flops_per_batch_tensor, reduce_operation='SUM')
```

Comment:
I thiiink there might be a `dist.all_reduce_object(..)` that could save you the `torch.tensor` stuff but don't worry about it unless you're confident that it's safe

Reply:
Only have an `all_gather_object` :(

I'm going to leave off for now because I don't want to rerun the tests ahead of release deadline, but this is probably a reasonable update we'd want to do in dist since this tensor wrapping occurs in a few places

Reply:
`flops_per_batch_tensor  = sum(all_gather_object(device_flops_per_batch))`